usaspending_api/download/v2/download_transaction_count.py

```diff
@@ -47,7 +48,11 @@ def post(self, request):
             "calculated_transaction_count": total_count,
             "maximum_transaction_limit": settings.MAX_DOWNLOAD_LIMIT,
             "transaction_rows_gt_limit": total_count > settings.MAX_DOWNLOAD_LIMIT,
-            "messages": [get_time_period_message()],
+            "messages": [
+                get_generic_filters_message(
+                    set(list(self.original_filters.keys())), [elem["name"] for elem in AWARD_FILTER]
```

Comment:
This got tagged by Code Climate as being duplicate code with the other places it's used. I'm reluctant to put this in a helper function for helper's sake, since this line's logic is very dependent on how the data is stored in this class. The helper would be tightly coupled against 4 view classes, which IMO is a worse issue than just having some boiler plate. Any thoughts?

Reply:
> since this line's logic is very dependent on how the data is stored in this class.

This leads to the question: why? Why couldn't the attributes be standardized in the classes? We could re-implement the parent class `APIView` to maintain the attributes. `get_generic_filters_message()` could then be run for all API responses and contingent on booleans in the child class (or some other flag/toggle/list)

Reply:
There is a lot of functionality that can be DRYed up in this classes. I'm concerned about going on TWO unrequested refactoring rampages this sprint though...

Reply:
You're the one who started the conversation 😄 I'm simply providing thoughts.

Reply:
Looks like this convo got a little sidetracked. Do you feel like this is a situation where it's safe to overrule CodeClimate for this PR, or should I be working on refactoring this?

Reply:
> Looks like this convo got a little sidetracked
How so? There is room for cleaning this up so CC is happy

> Do you feel like this is a situation where it's safe to overrule CodeClimate for this PR, or should I be working on refactoring this?
The repeated set(list(...)) and list comprehension isn't a nice line of code to duplicate. Is there an easy way to simplify the function call code? I'd be willing to override CC if the code isn't as messy